pkg/processor/openfaas.go

```diff
@@ -0,0 +1,117 @@
+package processor
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	sdk "github.com/openfaas-incubator/connector-sdk/types"
+	"github.com/openfaas/faas-provider/auth"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/connection"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vc-event-router/pkg/events"
+
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+const (
+	topicDelimiter     = ","
+	rebuildInterval    = time.Second * 10
+	timeout            = time.Second * 15
+	authMethodOpenFaaS = "basic_auth" // only this method is supported by the processor
+)
+
+// openFaaS implements the Processor interface
+type openFaaS struct {
+	controller sdk.Controller
+	source     string
+	verbose    bool
+}
+
+// NewOpenFaaSProcessor returns an OpenFaaS processor for the given stream
+// source. Asynchronous function invokation can be configured for
+// high-throughput (non-blocking) requirements.
+func NewOpenFaaSProcessor(cfg connection.Config, source string, verbose bool) (Processor, error) {
+	o := openFaaS{
```

Comment:
`o` -> `openFaaS` if the structure's name is changed to `OpenFaaS`

Reply:
also for the implementations where `o *openFaaS` -> `o *OpenFaaS`

Reply:
Another suggestion would be `o->openfaas`. But, I do think because it's a returned value, it should be a word.